traitsui/wx/button_editor.py

```diff
@@ -106,9 +109,14 @@ def init(self, parent):
             self.update_object, "clicked", dispatch="ui"
         )
         self.sync_value(self.factory.label_value, "label", "from")
-
+        self.sync_value(self.factory.image_value, "image", "from")
         self.set_tooltip()
 
+    @observe("image")
+    def _image_updated(self, event):
+        image = event.new
+        self._control.image = image
```

Comment:
This doesn't seem right - should it be `image.get_icon()` or `image.get_image()` or something like that?

Reply:
So for Qt, `image.create_icon()` returns a `QtGui.QIcon` which can then be passed as the argument of the `setIcon` method on the control.  

In this case on wx, the `_control` is a `pyface.ui.wx.image_button.ImageButton` and `control` is the control of that (ie `self.control = self._control.control`) which is a `wx.Window`.  Neither of which have "icon" related methods, etc.  `ImageButton` simply has an `image = Instance(ImageResource, allow_none=True)` trait, and so I set it directly to the `ImageResource` instance from the `image` trait here.

I agree it seems incorrect, and in fact it ends up working not exactly right, see the following screenshot:
<img width="336" alt="Screen Shot 2021-04-20 at 9 22 20 AM" src="https://user-images.githubusercontent.com/36972686/115431238-ee104f00-a1b9-11eb-965f-8418af329346.png">

`ImageButton` has an `_image_changed` method which sets 
```
self._img = image.create_image()
self._image = self._img.ConvertToBitmap()
```
which seems like maybe we would want to use create_icon instead or something?  I am not sure I was unable to figure this out.  There is code in `ImageButton.__init__` method which looks as though it is trying to figure out the size of the image and the size of the label (and that seems to work correctly - as in the above image looks as though it has the perfect button size to fit the image and label side by side).  However, it seems both the image and label are getting drawn at the center which likely isn't intentional (?)
I suspect this is might be a problem in `Imagebutton._on_paint`

Reply:
FWIW: I tried setting `orientation="horizontal"` and now I see this

<img width="336" alt="Screen Shot 2021-04-20 at 9 48 37 AM" src="https://user-images.githubusercontent.com/36972686/115434489-996ed300-a1bd-11eb-99cf-b72217f93e75.png">

I think we may be using some calculated layout value in the wrong place or something of the sort.  Looking into it now

Reply:
Also don't know how I'm just seeing this, but the dynamic label isn't working either!

Reply:
The label can be fixed by overwriting the `_label_changed` method in the `traitsui.wx.button_editor.CustomEditor` class to be `self._control.label = self.string_value(label)`.

Reply:
okay I have successfully fixed this locally.

<img width="336" alt="Screen Shot 2021-04-20 at 10 15 22 AM" src="https://user-images.githubusercontent.com/36972686/115437754-6d555100-a1c1-11eb-8481-591027a242a2.png">



I will push the changes to this PR and open a separate PR in pyface

Reply:
I've opened https://github.com/enthought/pyface/pull/932